apps/web/src/app/(protectedPage)/forms/[formId]/_components/formEditor/formEditorCard.tsx

```diff
@@ -172,6 +197,15 @@ export function FormEditorCard({ form }: Readonly<Props>) {
       }
     }
   };
+  const toggleRemoveField = (index: number) => {
+    const field = fields[index];
+    if (field) {
+      update(index, {
+        ...field,
+        isMarkedForRemoval: !field.isMarkedForRemoval,
+      } as { fieldName: string; isMarkedForRemoval: boolean });
+    }
+  };
```

Comment:
Use below code:

```
  const toggleRemoveField = (
    fieldIndex: number,
    field: { fieldName: string; id: string },
  ) => {
    // Check if field is already marked for removal
    if (removeFieldsIds.includes(field.id)) {
      setState((cs) => ({
        ...cs,
        removeFieldsIds: removeFieldsIds.filter((id) => id !== field.id),
      }));
      return;
    }

    const fieldValue = formHook.getValues().formFields[fieldIndex]?.fieldName;
    // check if field value is empty then we can safely remove it instead of marking it for removal
    if (fieldValue === "") {
      // remove field
      remove(fieldIndex);
      return;
    }

    // mark field for removal
    setState((cs) => ({
      ...cs,
      removeFieldsIds: [...removeFieldsIds, field.id],
    }));
  };
```